apps/challenges/views.py

```diff
@@ -340,3 +343,56 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_submissions_of_challenge(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns all the submissions for a particular challenge
+    """
+    # To check for the corresponding challenge from challenge_pk.
+    challenge_model = get_model_object(Challenge)
+    challenge = challenge_model(challenge_pk)
+
+    # To check for the corresponding challenge phase from the challenge_phase_pk and challenge.
+    try:
+        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase_pk, challenge=challenge)
+    except ChallengePhase.DoesNotExist:
+        response_data = {'error': 'Challenge Phase {} does not exist'.format(challenge_phase_pk)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
+
+    # To check for the user as a host of the challenge from the request and challenge_pk.
+    if is_user_a_host_of_challenge(user=request.user, challenge_pk=challenge_pk):
+        submissions = Submission.objects.filter(challenge_phase__challenge=challenge).order_by('-submitted_at')
+        paginator, result_page = paginated_queryset(submissions, request)
+        try:
+            serializer = ChallengeSubmissionManagementSerializer(result_page, many=True, context={'request': request})
+            response_data = serializer.data
+            return paginator.get_paginated_response(response_data)
+        except:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    # To check for the user as a participant of the challenge from the request and challenge_pk.
+    elif has_user_participated_in_challenge(user=request.user, challenge_id=challenge_pk):
+
+        # get participant team object for the user for a particular challenge.
+        participant_team_pk = get_participant_team_id_of_user_for_a_challenge(
+                request.user, challenge_pk)
+
+        submissions = Submission.objects.filter(participant_team=participant_team_pk,
```

Comment:
Yes , Currently the API doesn't contain the support for the query parameters for a challenge host like `challenge_phase or date` etc. But we will have that support in future. So, for now the submissions are filtered on the basis of `challenge` only. I had a talk regarding this with @deshraj.
Moreover, for the participants the results are being filtered on the basis of `challenge phase `that the participant selects from the dropdown menu on frontend.
Please tell if something is still not clear.